delivery_transsmart/models/res_config.py

```diff
@@ -108,47 +108,87 @@ def get_transsmart_service(self):
     def get_transsmart_carrier_tag(self):
         return self.env['ir.model.data'].get_object('delivery_transsmart', 'res_partner_category_transsmart_carrier')        
 
+    def _get_odata_filter(self, transsmart_ids):
+        """
+        assembling an odata $filter
+        http://www.odata.org/documentation/
+        odata-version-2-0/uri-conventions/
+
+        Gets all the records that their Id is not in transsmart_ids
+        """
+        _filter = ''
+        for index, code in enumerate(transsmart_ids):
+            if index < len(transsmart_ids) - 1:
```

Comment:
The if and the else logic are nearly the same,  except for the word and.  In these cases have the if else conditions fill a variable with ' and '  or '' and then make the rest one statement.